webauthn-rs-core/src/core.rs

```diff
@@ -470,56 +470,49 @@ impl WebauthnCore {
         // Now, match based on the attest_format
         debug!("attestation is: {:?}", &attest_format);
 
-        let credential: Credential = match attest_format {
-            AttestationFormat::FIDOU2F => verify_fidou2f_attestation(
-                acd,
-                &data.attestation_object,
-                &client_data_json_hash,
-                policy,
-            ),
-            AttestationFormat::Packed => verify_packed_attestation(
-                acd,
-                &data.attestation_object,
-                &client_data_json_hash,
-                policy,
-                req_extn,
-            ),
-            AttestationFormat::Tpm => verify_tpm_attestation(
-                acd,
-                &data.attestation_object,
-                &client_data_json_hash,
-                policy,
-                req_extn,
-            ),
+        let opt_req_extn: Option<&RequestRegistrationExtensions> = match attest_format {
+            AttestationFormat::Packed
+            | AttestationFormat::Tpm
+            | AttestationFormat::AppleAnonymous => Some(req_extn),
+            _ => None,
+        };
+
+        let attestation_data = match attest_format {
+            AttestationFormat::FIDOU2F => {
+                verify_fidou2f_attestation(acd, &data.attestation_object, &client_data_json_hash)
+            }
+            AttestationFormat::Packed => {
+                verify_packed_attestation(acd, &data.attestation_object, &client_data_json_hash)
+            }
+            AttestationFormat::Tpm => {
+                verify_tpm_attestation(acd, &data.attestation_object, &client_data_json_hash)
+            }
             AttestationFormat::AppleAnonymous => verify_apple_anonymous_attestation(
                 acd,
                 &data.attestation_object,
                 &client_data_json_hash,
-                policy,
-                req_extn,
             ),
-            AttestationFormat::None => {
-                verify_none_attestation(acd, &data.attestation_object, policy)
-            }
+            AttestationFormat::None => Ok(ParsedAttestationData::None),
             attest => {
                 if self.ignore_unsupported_attestation_formats {
                     debug!(?attest);
                     // No other types are currently implemented
                     Err(WebauthnError::AttestationNotSupported)
                 } else {
-                    let credential_public_key = COSEKey::try_from(&acd.credential_pk)?;
-                    Ok(Credential::new(
-                        acd,
-                        &data.attestation_object.auth_data,
-                        credential_public_key,
-                        policy,
-                        ParsedAttestationData::Uncertain,
-                        None,
-                    ))
+                    Ok(ParsedAttestationData::Uncertain)
                 }
             }
         }?;
 
+        let credential: Credential = Credential::new(
```

Comment:
Since we have the Credential::new now outside of the attestation code, Ithink this should be "after" the call to verify_attestation_ca_chain.

Reply:
Maybe even move it after the algorithm and excluded credentials check, construct it in last `Ok()` at the end?

Reply:
Yep, that's good by me!

Reply:
Ah I remember why I put it above verify_attestation_ca_chain, attestation_data would need to implement Copy or Credential should use a reference because it is used after move (use in verify_attestation_ca_chain, move in Credential::new). 
The other checks would also need to get the `credentials.*` properties from somewhere else if it were moved all the way to the result statement.